CAIPs/caip-x.md

```diff
@@ -0,0 +1,110 @@
+
+### Example Implementation
+
+- **Wallets**:
+
+```javascript
+function announceExtensionId() {
+  const providerInfo = {
+    name: 'Example Wallet',
+    icon: 'data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg"/>',
+    rdns: 'com.example.wallet',
+    extensionId: 'your_extension_id_here'
+  };
+  document.dispatchEvent(new CustomEvent('walletExtensionIdAnnouncement', {
+    detail: Object.freeze(providerInfo)
+  }));
+}
+
+announceExtensionId();
+
+window.addEventListener('requestWalletExtensionId', () => {
+  announceExtensionId();
+});
+```
+
+- **dApps**:
+
+```javascript
+window.addEventListener('walletExtensionIdAnnouncement', function(e) {
+  const providerInfo = e.detail;
+  // Use providerInfo.extensionId to establish communication
+});
+
+window.dispatchEvent(new Event("requestWalletExtensionId"));
+```
+
+**Content Script Utilization**:
+While this CAIP significantly advances the way wallets interact with dApps through the `externally_connectable` API, it still necessitates the use of a content script. This script, however, requires fewer permissions compared to traditional methods and can be injected synchronously to facilitate the initial communication process. This adjustment is necessary to bridge the current browser limitations and set the stage for future enhancements.
+
+- **Synchronous Injection**: The content script can be injected synchronously with the page load, ensuring that the wallet's extension ID is available as soon as the dApp begins its initialization process. This method avoids the delays and potential conflicts associated with asynchronous script loading, leading to a smoother and more predictable interaction pattern.
+    
+- **Security and Efficiency**: Although using a content script still presents a minimal footprint on the user's browsing experience, this approach is a compromise that balances current browser capabilities with the need for enhanced security and performance. By minimizing the script's functionality to only what is necessary for ID transmission, the risks associated with broader access permissions are mitigated.
+
+**Browser Manifest Adjustments**:
+
+Wallets can configure the `externally_connectable` section of their browser extension manifest to specify which domains are allowed to communicate with the extension. To avoid restricting communication to a pre-defined list of sites, wallets may use wildcards (e.g., `*.example.com`).
+
+**Backward Compatibility**:
+
+This protocol should be implemented in such a way that it does not interfere with existing implementations of EIP-1193 providers. Wallets may need to support both the new extension ID-based communication protocol and the traditional EIP-1193 provider interface.
```

Comment:
thoughts on adding a little blurb on how we can migrate away from this pattern when a browser native solutions arrives?